### PR TITLE
Change docker-clean build rule in Makefile

### PR DESCRIPTION
**Pull request by jjwatt**
- Use a make foreach macro and rmi instead of grep and xargs.

##### SUMMARY

Use a make foreach macro and rmi instead of grep and xargs.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Other (Makefile)

##### AWX VERSION

```
awx: 0.1.dev32848+g69cf85a
```

##### ADDITIONAL INFORMATION

Please comment on the `TODO` I added if you know the answer so that we can resolve it in the PR.

```

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -527,10 +527,11 @@ docker-compose-build-swagger: awx/projects docker-compose-sources
 SCHEMA_DIFF_BASE_BRANCH ?= devel
 detect-schema-change: genschema
 	curl https://s3.amazonaws.com/awx-public-ci-files/$(SCHEMA_DIFF_BASE_BRANCH)/schema.json -o reference-schema.json
-	# Ignore differences in whitespace with -b
+  # Ignore differences in whitespace with -b
 	diff -u -b reference-schema.json schema.json
 
 docker-compose-clean: awx/projects
+  # FIXME: Check for file first
```
Comment (AlanCoding) [inline Makefile]: I'm not really sure what this FIXME is suggesting. Could you clarify?

File: Makefile [hunk 2]
```diff
@@ -547,10 +548,9 @@ docker-compose-build:
 	    --cache-from=$(DEV_DOCKER_TAG_BASE)/awx_devel:$(COMPOSE_TAG) .
 
 docker-clean:
-	$(foreach container_id,$(shell docker ps -f name=tools_awx -aq && docker ps -f name=tools_receptor -aq),docker stop $(container_id); docker rm -f $(container_id);)
-	if [ "$(shell docker images | grep awx_devel)" ]; then \
-	  docker images | grep awx_devel | awk '{print $$3}' | xargs docker rmi --force; \
-	fi
+	-$(foreach container_id,$(shell docker ps -f name=tools_awx -aq && docker ps -f name=tools_receptor -aq),docker stop $(container_id); docker rm -f $(container_id);)
+  # TODO: Figure out if it was intentional for the old code to not remove receptor images.
```
Comment (AlanCoding) [inline Makefile]: The receptor images are only used for building images and for hop nodes. This obviously only attempts to deal with the awx_devel image. It also doesn't touch the redis or postgres image, because these are external and should change infrequently. While receptor is more integral and picking up changes are more important, you get this by refreshing the awx_devel image, itself... unless you're building locally or using hop nodes.

I think I agree that in some context it would be appropriate to remove receptor images, but the distinction between `make docker-compose-clean` and `make docker-clean` is, itself, challenging to undertand.

File: Makefile (hunk 1 above)
Comment (jjwatt) [reply Makefile]: Oh yeah. If the `docker-compose.yml` file isn't there, you get kind of a nasty error. But, this FIXME could apply to a ton of these rules/commands, so it might be better to just leave it out for now.

File: Makefile (hunk 2 above)
Comment (jjwatt) [reply Makefile]: Thanks. And, I agree about those rules being challenging to understand. @TheRealHaoLiu and I have been talking about fixing up our GNU Make rules and de-complicating them.

File: Makefile [hunk 3]
```diff
@@ -527,7 +527,7 @@ docker-compose-build-swagger: awx/projects docker-compose-sources
 SCHEMA_DIFF_BASE_BRANCH ?= devel
 detect-schema-change: genschema
 	curl https://s3.amazonaws.com/awx-public-ci-files/$(SCHEMA_DIFF_BASE_BRANCH)/schema.json -o reference-schema.json
-	# Ignore differences in whitespace with -b
+  # Ignore differences in whitespace with -b
```
Comment (TheRealHaoLiu) [inline Makefile]: [nit] is this change intentional

Comment (jjwatt) [reply Makefile]: Sort of. Can you tell if those are spaces or tabs in the other one? I have a habit of using spaces for "make" stuff and only using tabs for "command" blocks because anything that begins with tabs will be passed to the shell. Maybe it's just more spaces than my editor converted it to, or maybe it's a tab that I converted to spaces. I'll check.

Comment (jjwatt) [reply Makefile]: Went ahead and just reversed that hunk

**Review by AlanCoding — APPROVED** [review]
updated form looks more consistent and direct
